.travis.yml

```diff
@@ -1,27 +1,127 @@
+######## Testing minimal compiler requirements ########
+# GCC          6.1.0        here: 6.4.0
+# Clang        4.0          here: 3.8,3.9 not compiling
+# Apple Clang  9            here: clang-900.0.39.2
+#######################################################
+
 language: cpp
 sudo: false
-os: linux
-dist: trusty
-compiler: gcc
 
-addons:
-    apt:
-        sources:
+matrix:
+  include:
+########################## OS X #########################
+## test gcc6 - single node/rank with threading backend ##
+    - os: osx
+      osx_image: xcode9.2
+      env:
+        - MATRIX_EVAL="brew install gcc@6 && brew link --force --overwrite gcc@6 && CC=gcc-6 && CXX=g++-6"
+        - BUILD_NAME=cthread WITH_DISTRIBUTED=serial
+      compiler: gcc-6
+
+## test gcc6 - mpi with threading backend ##
+    - os: osx
+      osx_image: xcode9.2
+      env:
+        - MATRIX_EVAL="brew install gcc@6 && brew link --force --overwrite gcc@6 && CC=gcc-6 && CXX=g++-6"
```

Comment:
Do these tests for gcc 6 take a long time to run by virtue of having to install gcc via homebrew?

Reply:
In total, it takes roughly 8 min. The installing of gcc6 takes only 15s (according to Travis)

Reply:
Great!  Whenever I have installed gcc on my mac using Homebrew it has compiled gcc, which took a loooong time (admittedly the last time I did this was years ago).